### PR TITLE
[Documentation][Book] Translations

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no |
| New docs?    | yes |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: docs/book/architecture/translations.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        */
+       public function getName()
+       {
+           return $this->name;
+       }
+
+       /**
+        * @param string $name
+        */
+       public function setTitle($name)
```
Comment (GSadee) [inline docs/book/architecture/translations.rst]: `setName`

File: docs/book/architecture/translations.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+   }
+
+The ``getTranslation()`` method gets a translation for the current locale, while we are in the shop, but we can also manually
+impose the locale - ``getTranslation('pl_PL')`` will return a polish translation.
+
+How to add a new translation programmatically?
+----------------------------------------------
+
+You can programmatically add a translation to any of the translatable resources in Sylius.
+Let's see how to do it on the exaple of a ProductTranslation.
```
Comment (GSadee) [inline docs/book/architecture/translations.rst]: example

File: docs/book/architecture/translations.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+.. code-block:: php
+
+   // find a product to add a translation to it
+
+   /** @var ProductInterface $product */
+   $product = $this->container->get('sylius.repository.product')->findOneBy(['name' => 'Radiohead Mug']);
+
+   // Create a new translation of product, give it a translated name and slugin the chosen locale
```
Comment (GSadee) [inline docs/book/architecture/translations.rst]: slug in

File: docs/book/architecture/translations.rst [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+   /** @var ProductTranslation $translation */
+   $translation = new ProductTranslation();
+
+   $translation->setLocale('pl_PL');
+   $translation->setName('Kubek Radiohead');
+   $translation->setSlug('kubek-radiohead');
+
+
```
Comment (GSadee) [inline docs/book/architecture/translations.rst]: Unnecessary blank line

File: docs/book/architecture/translations.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        * @param string $name
+        */
+       public function setName($name)
+       {
+           $this->getTranslation()->setName($name);
+       }
+   }
+
+The ``getTranslation()`` method gets a translation for the current locale, while we are in the shop, but we can also manually
+impose the locale - ``getTranslation('pl_PL')`` will return a polish translation.
```
Comment (NeverResponse) [inline docs/book/architecture/translations.rst]: Missing info what will be returned if there is no translation for passed locale.

Comment (TheMadeleine) [inline docs/book/architecture/translations.rst]: Good point! :+1:

File: docs/book/architecture/translations.rst [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+   single: Translations
+
+Translations
+============
+
+Sylius uses the approach of personal translations - where each entity is bound with a translation entity, that has it's
+own table (instead of keeping all translations in one table for the whole system).
+This results in having the ``ProductTranslation`` class and ``sylius_product_translation`` table for the ``Product`` entity.
+
+The *engine* for handling translations in Sylius is the **ResourceBundle**
```
Comment (michalmarcinkowski) [inline docs/book/architecture/translations.rst]: `engine` -> `logic` wouldn't be a better word?

File: docs/book/architecture/translations.rst [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+You can programmatically add a translation to any of the translatable resources in Sylius.
+Let's see how to do it on the example of a ProductTranslation.
+
+.. code-block:: php
+
+   // Find a product to add a translation to it
+
+   /** @var ProductInterface $product */
+   $product = $this->container->get('sylius.repository.product')->findOneBy(['name' => 'Radiohead Mug']);
```
Comment (michalmarcinkowski) [inline docs/book/architecture/translations.rst]: `findOneBy(['name'` will not work 😢  the `name` is on `ProductTranslation` so we need a join as well as include the locale, what is more the name doesn't need to be unique, so this query can throw exception. 

There is `findByName($name, $locale)` or we can `findOneBy(['code`.

Comment (TheMadeleine) [inline docs/book/architecture/translations.rst]: True 👍

**Comment by TheMadeleine** [comment]
@michalmarcinkowski fixed now ;)

File: docs/book/architecture/translations.rst [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+   single: Translations
+
+Translations
+============
+
+Sylius uses the approach of personal translations - where each entity is bound with a translation entity, that has it's
+own table (instead of keeping all translations in one table for the whole system).
+This results in having the ``ProductTranslation`` class and ``sylius_product_translation`` table for the ``Product`` entity.
+
+The logic of handling translations in Sylius is the **ResourceBundle**
```
Comment (michalmarcinkowski) [inline docs/book/architecture/translations.rst]: > The logic ... in Sylius is **in** the ResourceBundle

or restore the previous sentence :)

> The **engine** ... is the ResourceBundle

File: docs/book/architecture/translations.rst [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+You can programmatically add a translation to any of the translatable resources in Sylius.
+Let's see how to do it on the example of a ProductTranslation.
+
+.. code-block:: php
+
+   // Find a product to add a translation to it
+
+   /** @var ProductInterface $product */
+   $product = $this->container->get('sylius.repository.product')->findOneByCode(['code' => 'radiohead-mug-code']);
```
Comment (michalmarcinkowski) [inline docs/book/architecture/translations.rst]: findOneBy~~Code~~(['code' 😄

Comment (TheMadeleine) [inline docs/book/architecture/translations.rst]: 🎉

File: docs/book/architecture/translations.rst [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+   <?php
+
+   namespace AppBundle\Entity;
+
+   use Sylius\Component\Resource\Model\TranslatableInterface;
+   use Sylius\Component\Resource\Model\TranslatableTrait;
+
+   class Supplier implements TranslatableInterface
+   {
+       use TranslatableTrait;
```
Comment (NeverResponse) [inline docs/book/architecture/translations.rst]: The translations collection needs to be initialised in the supplier's constructor.

Comment (TheMadeleine) [inline docs/book/architecture/translations.rst]: Done now! :)

**Comment by pjedrzejewski** [comment]
Thanks Magda and everyone for review, nice work!

**Pull request merged**